storage_service/common/tests/test_command_import_aip.py

```diff
@@ -0,0 +1,113 @@
+import os
+import uuid
+
+import pytest
+from django.core.management import call_command
+from locations import models
+
+TEST_DIR = os.path.dirname(os.path.realpath(__file__))
+FIXTURES_DIR = os.path.join(TEST_DIR, "fixtures")
+AIP_PATH = os.path.join(FIXTURES_DIR, "import_aip_test.7z")
+
+
+@pytest.fixture
+def aip_location(db, tmp_path):
+    space_temp = tmp_path / "sub"
+    space_temp.mkdir()
+    space = models.Space.objects.create(
+        uuid=str(uuid.uuid4()),
+        path=space_temp,
+        access_protocol=models.Space.LOCAL_FILESYSTEM,
+        staging_path=space_temp,
+    )
+    pipeline = models.Pipeline.objects.create(uuid=str(uuid.uuid4()))
+    aipstore = models.Location.objects.create(
+        uuid=str(uuid.uuid4()),
+        space=space,
+        purpose="AS",
+        relative_path="",
+    )
+    models.Location.objects.create(
+        space=space, purpose=models.Location.STORAGE_SERVICE_INTERNAL, relative_path=""
+    )
+    models.LocalFilesystem.objects.create(space=space)
+    models.LocationPipeline.objects.get_or_create(pipeline=pipeline, location=aipstore)
+    return aipstore
+
+
+@pytest.mark.django_db
+def test_import_aip(mocker, capsys, aip_location):
+    mocker.patch("os.chown")
+    mocker.patch("pwd.getpwnam")
+    call_command(
+        "import_aip",
+        "--decompress-source",
+        "--aip-storage-location",
+        aip_location.uuid,
+        AIP_PATH,
+    )
+    captured = capsys.readouterr()
+    # shutil.rmtree(AIP_PATH.split("import_aip_bag")[-1])
+    assert "Successfully imported AIP" in captured.out
+
+
+# Test if package is compressed
+@pytest.mark.django_db
+def test_import_aip_compressed(mocker, capsys, aip_location):
+    mocker.patch("os.chown")
+    mocker.patch("pwd.getpwnam")
+    call_command(
+        "import_aip",
+        "--decompress-source",
+        "--aip-storage-location",
+        aip_location.uuid,
+        AIP_PATH,
+        "--compression-algorithm",
+        "7z with bzip",
+    )
+    captured = capsys.readouterr()
+    assert "Successfully imported AIP" in captured.out
+
+    # Verify a new package was created.
+    assert models.Package.objects.count() == 1
+
+    # Verify package is not compressed.
+    package = models.Package.objects.first()
+    assert package.is_compressed
+
+
+# Test if package is compressed or uncompressed
+@pytest.mark.parametrize(
+    "compression_algorithm, expected_compression_algorithm",
+    [
+        # matching for compressed package
```

Comment:
Could these comments be full sentences? It's a bit hard to understand them.

Reply:
# matching compression algorithm flag if package is compressed

Will this work?

Reply:
sure let me change it